src/sentry_envelope.c

```diff
@@ -400,75 +318,41 @@ prepare_http_request(const sentry_uuid_t *event_id,
 
     h = &rv->headers[rv->headers_len++];
     h->key = "content-type";
-    h->value = sentry__string_clone(content_type);
+    h->value = sentry__string_clone(ENVELOPE_MIME);
 
     h = &rv->headers[rv->headers_len++];
     h->key = "content-length";
     h->value = sentry__int64_to_string((int64_t)payload_len);
 
-    switch (endpoint_type) {
-    case ENDPOINT_TYPE_ENVELOPE:
-        rv->url = sentry__dsn_get_envelope_url(&options->dsn);
-        break;
-    case ENDPOINT_TYPE_STORE:
-        rv->url = sentry__dsn_get_store_url(&options->dsn);
-        break;
-    case ENDPOINT_TYPE_MINIDUMP:
-        rv->url = sentry__dsn_get_minidump_url(&options->dsn);
-        break;
-    case ENDPOINT_TYPE_ATTACHMENT: {
-        rv->url = sentry__dsn_get_attachment_url(&options->dsn, event_id);
-        break;
-    }
-    default:
-        rv->url = NULL;
-    }
-
-    rv->payload = payload;
-    rv->payload_len = payload_len;
-    rv->payload_owned = payload_owned;
+    rv->body = body;
+    rv->body_len = payload_len;
+    rv->body_owned = body_owned;
 
     return rv;
 }
 
-static void
-gen_boundary(char *boundary)
-{
-#if SENTRY_UNITTEST
-    sentry_uuid_t boundary_id
-        = sentry_uuid_from_string("0220b54a-d050-42ef-954a-ac481dc924db");
-#else
-    sentry_uuid_t boundary_id = sentry_uuid_new_v4();
-#endif
-    sentry_uuid_as_string(&boundary_id, boundary);
-    strcat(boundary, "-boundary-");
-}
-
 void
 sentry__envelope_for_each_request(const sentry_envelope_t *envelope,
     bool (*callback)(sentry_prepared_http_request_t *,
         const sentry_envelope_t *, void *data),
     const sentry_rate_limiter_t *rl, void *data)
 {
     sentry_prepared_http_request_t *req;
-    sentry_uuid_t event_id = sentry__envelope_get_event_id(envelope);
 
     if (envelope->is_raw) {
-        req = prepare_http_request(&event_id, ENDPOINT_TYPE_ENVELOPE,
-            "application/x-sentry-envelope", envelope->contents.raw.payload,
+        req = prepare_http_request(envelope->contents.raw.payload,
```

Comment:
Would this function make more sense in a base transport file rather than `envelope.c`? It seems that most of what's happening here is transport-specific. See comments below regarding splitting this up.

Reply:
both envelope and envelope_item are private to that file right now. Not quite sure if we should expose it right now?

Reply:
Either way, we can do this incrementally anyway. First making a breaking change to hide the transport struct, and then expose the http request interface at some point later.

Reply:
Okay, that sounds good to me. 

Eventually we might have to expose the envelope for the sake of custom transports. At that, it might be nice to have the two functions separate.